### PR TITLE
refactor delayCodes subscription

### PR DESCRIPTION
**Pull request by Abdiiir**

### CODE REVIEW

File: src/app/view/timestamp-editor/timestamp-editor.component.ts [hunk 1]
```diff
@@ -62,6 +61,7 @@ export class TimestampEditorComponent implements OnInit {
   respondingToTimestampInfo: TimestampInfo;
   TimestampResponseStatus = TimestampResponseStatus;
   fullVesselDetails: Vessel;
+  delayCodes: Observable<DelayCode[]>;
```
Comment (nt-as) [inline src/app/view/timestamp-editor/timestamp-editor.component.ts:64]: Use `delayCodes$` (convention in Angular to signal it is an Observable)

File: src/app/view/transport-call-creator/transport-call-creator.component.ts [hunk 2]
```diff
@@ -42,18 +43,16 @@ export class TransportCallCreatorComponent implements OnInit {
...
   eventTimestampTime: string;
   timestampDefinitions: TimestampDefinitionTO[] = [];
   timestampTypes: SelectItem[] = [];
   delayCodeOptions: SelectItem[] = [];
-  delayCodes: DelayCode[];
   publisherRoleOptions: SelectItem[] = [];
   publisherRoles: PublisherRole[] = [];
   timestampChecking: boolean;
   locationNameLabel: string;
+  delayCodes: Observable<DelayCode[]>
```
Comment (nt-as) [inline src/app/view/transport-call-creator/transport-call-creator.component.ts:55]: Use `delayCodes$` (convention in Angular to signal it is an Observable)

File: src/app/view/timestamp-editor/timestamp-editor.component.ts [hunk 3]
```diff
@@ -98,7 +98,7 @@ export class TimestampEditorComponent implements OnInit {
       vesselPositionLatitude: new FormControl(null, [Validators.pattern("^[0-9.]*$"), Validators.maxLength(10)]),
       milesToDestinationPort: new FormControl(null, [Validators.pattern('^[0-9]+(.[0-9]?)?$')]),
       remark: new FormControl(null),
-      delayCode: new FormControl({ value: '' }),
+      delayCodes$: new FormControl({ value: '' }),
```
Comment (nt-as) [inline src/app/view/timestamp-editor/timestamp-editor.component.ts:101]: This is not an observable (but a `FormControl`), so I do not think this should be have a `$` in its name.

Comment (Abdiiir) [reply src/app/view/timestamp-editor/timestamp-editor.component.ts:101]: That's a typo, will fix it

**Review by nt-as — APPROVED** [review]

**Pull request merged**